### ISSUE TITLE
Sqlite.FunctionalTests fail - Sqlite3 bad image format

### ISSUE THREAD

**Issue by CADbloke** [dotnet/efcore#4254]
**Using in VS2015pro .NET 4.5.1 x64 Win10, R# 10.2 test runner with [XUnit support](https://github.com/xunit/resharper-xunit)**

Sqlite DLL version 3.9.2

Got 158 failed tests running `EntityFramework.Sqlite.FunctionalTests` complaining that Sqlite3 was unable to load. It has obviously mixed up x64 / x84.
Related to #3720, distant cousin of #1074.

I am planning to use EF7 + Sqlite in a Win 7+ WPF .NET 4.5 app. I'm stuck with .NET 4.5 because I am using the [AutoCAD SDK](https://knowledge.autodesk.com/support/autocad/troubleshooting/caas/sfdcarticles/sfdcarticles/System-requirements-for-AutoCAD-2016.html)

Here's one of the exception messages - they all looked the same

```
System.AggregateException

One or more errors occurred.

Exception doesn't have a stacktrace

System.BadImageFormatException

An attempt was made to load a program with an incorrect format. (Exception from HRESULT: 0x8007000B)

   at Microsoft.Data.Sqlite.Interop.NativeMethods.Sqlite3_sqlite3.sqlite3_open_v2(IntPtr filename, Sqlite3Handle& ppDb, Int32 flags, IntPtr vfs)
   at Microsoft.Data.Sqlite.Interop.NativeMethods.sqlite3_open_v2(String filename, Sqlite3Handle& ppDb, Int32 flags, String vfs)
   at Microsoft.Data.Sqlite.SqliteConnection.Open()
   at Microsoft.Data.Entity.Sqlite.FunctionalTests.SqliteTestStore.CreateAndOpenConnection(Boolean sharedCache) in C:\Codez\z.Libraries\dotNET\EntityFramework7\test\EntityFramework.Sqlite.FunctionalTests\SqliteTestStore.cs:line 65
   at Microsoft.Data.Entity.Sqlite.FunctionalTests.SqliteTestStore.CreateShared(Action initializeDatabase) in C:\Codez\z.Libraries\dotNET\EntityFramework7\test\EntityFramework.Sqlite.FunctionalTests\SqliteTestStore.cs:line 47
   at Microsoft.Data.Entity.Sqlite.FunctionalTests.SqliteTestStore.GetOrCreateShared(String name, Action initializeDatabase) in C:\Codez\z.Libraries\dotNET\EntityFramework7\test\EntityFramework.Sqlite.FunctionalTests\SqliteTestStore.cs:line 18
   at Microsoft.Data.Entity.Sqlite.FunctionalTests.TestModels.SqliteNorthwindContext.GetSharedStore() in C:\Codez\z.Libraries\dotNET\EntityFramework7\test\EntityFramework.Sqlite.FunctionalTests\TestModels\SqliteNorthwindContext.cs:line 17
   at Microsoft.Data.Entity.Sqlite.FunctionalTests.NorthwindQuerySqliteFixture..ctor() in C:\Codez\z.Libraries\dotNET\EntityFramework7\test\EntityFramework.Sqlite.FunctionalTests\NorthwindQuerySqliteFixture.cs:line 20

Xunit.Sdk.TestClassException

The following constructor parameters did not have matching fixture data: NorthwindQuerySqliteFixture fixture

Exception doesn't have a stacktrace
```

Note - they all pass when building via the command line, as you would expect.

```
info: Exec
info:   program: cmd
info:   commandline: /C dnx test
info:   workingdir: test\EntityFramework.Sqlite.FunctionalTests
xUnit.net DNX Runner (32-bit DNX 4.5.1)
  Discovering: EntityFramework.Sqlite.FunctionalTests
  Discovered:  EntityFramework.Sqlite.FunctionalTests
  Starting:    EntityFramework.Sqlite.FunctionalTests
  Finished:    EntityFramework.Sqlite.FunctionalTests
=== TEST EXECUTION SUMMARY ===
   EntityFramework.Sqlite.FunctionalTests  Total: 1403, Errors: 0, Failed: 0, Sk
ipped: 0, Time: 49.379s
```

**Comment by natemcmaster** [comment]
In R#, you need to switch to x86. Our build process has a task that automatically copies sqlite3.dll into the `bin/Debug/` directory of the sqlite test projects. It chooses x86 because most of us on our team use TestDriven.NET, which executes in x86.

See https://github.com/aspnet/EntityFramework/blob/dev/makefile.shade#L47

**Comment by CADbloke** [comment]
Ah ha. Thanks very much for the explanation and clarification. I'm assuming that since it is all built as MSIL then it will run with both x64 and x86 Sqlite native DLLs. I will continue to build and test my local version like I'm supposed to, from the command line. btw, discovering that process wasn't easy. I shall probably use the packages from https://www.myget.org/gallery/aspnetvnext anyways,

FYI, one thing I noticed with the Sqlite v 3.10.0 download from sqlite.org - I compared the x64 & x86 `sqlite3.def` files and there were some differences.
x64 has `sqlite3_data_directory`, `sqlite3_fts5_may_be_corrupt`, `sqlite3_temp_directory` & `sqlite3_version` that the x86 version doesn't have. Nothing was missing from x64, I don't know if any of this is significant or not.
